validation/custom_constraint.rst

```diff
@@ -94,6 +94,17 @@ message as its argument and returns an instance of
 :class:`Symfony\\Component\\Validator\\Violation\\ConstraintViolationBuilderInterface`.
 The ``addViolation()`` method call finally adds the violation to the context.
 
+If the ``validate`` method is called with an incorrect ``$constraint``, an
```

Comment:
I think we could just add an inline comment in the code like this:

```php
if (!is_string($value)) {
    // throw this exception if your validator cannot handle the passed type so that it can marked as invalid
    throw new UnexpectedValueException($value, 'string');
}
```

What do you think?

Reply:
Maybe, but then we would lose the part about multiple types, or the comment would get very large.
Adding both the comment and the text with more explanation could work. What do you think?

Reply:
We can add a commented example for that:

```php
if (!is_string($value)) {
    // throw this exception if your validator cannot handle the passed type so that it can marked as invalid
    throw new UnexpectedValueException($value, 'string');

    // separate multiple types using pipes
    // throw new UnexpectedValueException($value, 'string|int');
}
```

Reply:
Ah, that's a good idea. I'll make the changes. Thank you, @xabbuh.